### ISSUE TITLE
Dropout Training vs Test

### ISSUE THREAD

**Issue by joaogui1** [deepmind/haiku#5]
I was looking at Haiku's dropout [implementation](https://github.com/deepmind/haiku/blob/master/haiku/_src/basic.py#L252) and it seems like there's no difference between the use of dropout during training and during test, yet normally dropout is only done [during training](https://github.com/tensorflow/tensorflow/blob/v2.1.0/tensorflow/python/keras/layers/core.py#L181), is this difference intended behavior?

**Comment by trevorcai** [comment]
Hey, thanks for the issue. This is intentional, as it mirrors the [`tf.nn.dropout`](https://www.tensorflow.org/api_docs/python/tf/nn/dropout) API.

People typically either:
- Set dropout_rate = 0 during test
- Do `if training: x = hk.dropout(rng, x, rate=...)`.

**Comment by joaogui1** [comment]
Got it, thanks!
